### PR TITLE
BookComponent

### PR DESCRIPTION
**Pull request by ranaagouda**

### CODE REVIEW

File: src/components/UI/BookComponent/BookComponent.module.css [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  width: 100%;
+  position: relative;
+}
+.content {
+  max-width: 100%;
+  gap: 4px;
+}
+.title {
+  color: var(--dark-primary-color);
+  text-overflow: ellipsis;
```
Comment (AdhamUsama25) [inline src/components/UI/BookComponent/BookComponent.module.css:20]: for the ellipsis to be active the item must have a max width with pixels, did you try it?

File: src/components/UI/BookComponent/BookComponent.module.css [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  white-space: nowrap;
+}
+.author {
+  color: var(--dark-gray-color);
+  text-overflow: ellipsis;
+  overflow: hidden;
+  white-space: nowrap;
+}
+
+.addButton {
```
Comment (AdhamUsama25) [inline src/components/UI/BookComponent/BookComponent.module.css:31]: adding transform duration will make it smother

File: src/components/UI/BookComponent/BookComponent.module.css [hunk 3]
```diff
@@ -0,0 +1,49 @@
+.book {
+  display: flex;
+  width: 128px;
+  flex-direction: column;
+  box-shadow: none;
```
Comment (AdhamUsama25) [inline src/components/UI/BookComponent/BookComponent.module.css:5]: this is already mentioned in the sx of the component

File: src/components/UI/BookComponent/BookComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  const handleMouseEnter = () => setIsHovering(true);
+  const handleMouseLeave = () => setIsHovering(false);
+
+  const handleClick = () => navigate(`/books/${book.id}`);
+
+  return (
+    <Card
+      className={classes.book}
+      style={{ boxShadow: "none" }}
+      onMouseEnter={handleMouseEnter}
```
Comment (AdhamUsama25) [inline src/components/UI/BookComponent/BookComponent.tsx:30]: I guess there is no need for this as CSS managed the hovering already

File: src/components/UI/BookComponent/BookComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      onMouseLeave={handleMouseLeave}
+      onClick={handleClick}
+    >
+      <CardMedia
+        className={classes.cover}
+        component="img"
+        image={book.cover}
+        alt={book.name}
+      />
+      {isHovering && (
```
Comment (AdhamUsama25) [inline src/components/UI/BookComponent/BookComponent.tsx:40]: Again you may remove the optional rendering and in CSS add `visibility: hidden` for the overlay button and `visibility: visible` for `.card: hover .addButton`

File: src/components/UI/BookComponent/BookComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        <Button
+          className={classes.addButton}
+          style={{
+            position: "absolute",
+            transition: "opacity 0.3s ease-in-out",
```
Comment (AdhamUsama25) [inline src/components/UI/BookComponent/BookComponent.tsx:45]: I see you made the transition duration here, please remove the redundancy in the CSS or remove it from here and depend on CSS (recommended)

File: src/components/UI/BookComponent/BookComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            xmlns="http://www.w3.org/2000/svg"
+            viewBox="0 0 448 512"
+            className={classes.plus}
+          >
+            <path d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z" />
+          </svg>{" "}
+        </Button>
+      )}
+
+      <CardContent className={classes.content} style={{ padding: "0" }}>
```
Comment (AdhamUsama25) [inline src/components/UI/BookComponent/BookComponent.tsx:59]: if possible ad the style to the class

File: src/components/UI/BookComponent/BookComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        <Typography
+          className={classes.title}
+          style={{
+            fontWeight: "700",
```
Comment (AdhamUsama25) [inline src/components/UI/BookComponent/BookComponent.tsx:63]: add style to class

File: src/components/UI/BookComponent/dummyBooks.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { Book } from "../../../Types/bookTypes";
+
+export const dummyBooks: Book[] = [
```
Comment (AdhamUsama25) [inline src/components/UI/BookComponent/dummyBooks.ts:3]: merge the main branch, remove this file, and use the dummy data provided

**Review by AdhamUsama25 — CHANGES_REQUESTED** [review]

File: src/components/UI/BookComponent/BookComponent.tsx (hunk 7 above)
Comment (ranaagouda) [reply src/components/UI/BookComponent/BookComponent.tsx:59]: it was there but mui styles overrode it

File: src/components/UI/BookComponent/BookComponent.tsx (hunk 8 above)
Comment (ranaagouda) [reply src/components/UI/BookComponent/BookComponent.tsx:63]: it was there but mui styles overrode it

File: src/components/UI/BookComponent/BookComponent.module.css (hunk 1 above)
Comment (ranaagouda) [reply src/components/UI/BookComponent/BookComponent.module.css:20]: No, I didn't but the ellipsis worked fine, I'll add it

File: src/components/UI/BookComponent/BookComponent.tsx (hunk 5 above)
Comment (ranaagouda) [reply src/components/UI/BookComponent/BookComponent.tsx:40]: I added them now
however, yesterday when I removed the optional rendering the component didn't seem clickable at all

File: src/components/UI/BookComponent/BookComponent.tsx (hunk 6 above)
Comment (ranaagouda) [reply src/components/UI/BookComponent/BookComponent.tsx:45]: it was there but mui styles overrode it

File: src/components/UI/BookComponent/dummyBooks.ts (hunk 9 above)
Comment (ranaagouda) [reply src/components/UI/BookComponent/dummyBooks.ts:3]: no dummy data for books is provided in dummydata.ts
I'll add this book data to the file ok

**Pull request merged**